og/bump/action.yml

```
@@ -0,0 +1,33 @@
+name: bump
+description: |
+  Automatically bump the version file of the OG repo
+
+inputs:
+  base_branch:
+    description: The base branch of the PR.
+    required: true
+  pr_body:
+    description: OG formatted body of the PR description.
+    required: true
+  pr_title:
+    description: SC formatted title of the PR.
+    required: true
+  dev_branch:
+    description: The dev branch of this action to connect to the workflow
+    required: false
+    default: 'main'
+
+runs:
+  using: "composite"
+
+  steps:
+    - run: python3 -m pip install semver
+      shell: bash
+    
+    - run: |
+        export DEFAULT_BRANCH="${{ inputs.base_branch }}"
+        export PR_BODY="${{ inputs.pr_body }}"
+        export PR_TITLE="${{ inputs.pr_title }}"
+        python -c "$(wget -qSO- https://raw.githubusercontent.com/XanaduAI/cloud-actions/${{ inputs.dev_branch }}/og/bump/tagger.py 2>/dev/null)" 
```

note this line in the review please

---

Because...?

---

That's really the reassurance I needed 😄

---

Just because I have to `wget` a file that lives in the same repo and pass the contents directly to `python -c`.
Alternatively, I can also do:
```suggestion
        python -c "$(git show origin/${{ inputs.dev_branch }}:og/bump/tagger.py" 
```

---

so the shared github action doesn't check out its own repo at all?

---

if this is really necessary, should this not happen in a separate line (creating a local file)? That way we would be able to see stderr in case something goes wrong?

---

We can, we just have to add it to `.gitignore` in every repo in which the workflow uses this action

---

can't we
- download the file to /tmp
- or delete it after using it?

---

Oh btw, we can see the `stderr` from the python process, the `wget` command will put some metadata in stderr which I'm ignoring 🤔